PL-38935-JIT-user-provisioning (#2137)
harness/developer-hub: docs/platform/3_User-Management/provision-use-jit.md

---
title: Provision users JIT
description: Just-in-time user provisioning.
sidebar_position: 2
---

```mdx-code-block
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
```

:::important
Currently, this feature is behind the feature flag `PL_ENABLE_JIT_USER_PROVISION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
:::

Just-in-time (JIT) provisioning in Harness lets you provision users automatically upon first sign in to Harness through SAML SSO.

> You can automatically provision users added to your SAML applications configured in Harness and allow them to access Harness.

Reviewer: I can't tell if it's the users that are configured in Harness, or the SAML apps that are configured in Harness. Also "them" is unclear... though I'm guessing it's the users.

Author: Rephrased.

Provisioning users automatically eliminates repetitive tasks related to manual provisioning, simplifying user management.

:::important

> Harness supports Just-in-time provisioning only if a new user attempts to log in via IDP.

Reviewer: ```suggestion
Harness supports JIT provisioning only if a new user attempts to log in via IDP.
```

:::

## Enable JIT provisioning in Harness

To enable JIT provisioning in Harness: 
1. Select **ACCOUNT SETTINGS**, and then **Authentication**.
2. Select **SAML Provider**.
   The **Add SAML Provider** settings appear.
3. In **Name**, enter a name for the SAML provider.
4. In **Select a SAML Provider**, select one of the following: 
   - Azure
   - Okta
   - OneLogin
   - Other

>    

Reviewer: This is probably not a good use of tabs. You can simply add the link to instructions to the bullet list above. This is easier for the user, and more importantly here reduces the cost of maintenance of this doc (having tabs unnecessarily adds to maintenance time... important to consider when we're very short on resources).

Author: Done.

   <Tabs>
      <TabItem value="Azure">
   ```

   For steps to configure, go to [SAML SSO with Azure](/docs/platform/Authentication/single-sign-on-saml#saml-sso-with-azure).

   ```mdx-code-block
      </TabItem>
      <TabItem value="Okta">
   ```   

   For steps to configure, go to [SAML SSO with Okta](/docs/platform/Authentication/single-sign-on-saml#saml-sso-with-okta).

   ```mdx-code-block
      </TabItem>
      <TabItem value="OneLogin">
   ```

   For steps to configure, go to [SAML SSO with OneLogin](/docs/platform/Authentication/single-sign-on-saml#saml-sso-with-onelogin).

   ```mdx-code-block
      </TabItem>
      <TabItem value="Others">
   ``` 

   For steps to configure, go to [SAML SSO with Keycloak](/docs/platform/Authentication/single-sign-on-saml#saml-sso-with-keycloak).

   ```mdx-code-block
      </TabItem>
   </Tabs>
   ```

5. Select **Enable JIT Provisioning**. 
   This enables Just-In-Time (JIT) provisioning to add users to Harness on their first login via the configured SAML provider.

6. To control who can get added to Harness on their first login, define the key-value that must be present in the SAML assertions by entering the following: 

>    - **JIT Validation Key**: To log in for the first time, the SAML assertion must contain the key.

Reviewer: ```suggestion
   - **JIT Validation Key**: The SAML assertion must contain this key on first login.
```

>    - **JIT Validation Value**: At the time of first login, enter the value for the validation key that must be present in the SAML assertion.

Reviewer: ```suggestion
   - **JIT Validation Value**: The value for the validation key that must be present in the SAML assertion on first login.
```

>      ![](./static/jit-user-provisioning.png)

Reviewer: I think our convention is to have the image you might be referring to before the instruction on what to do. So if following this, this image should be in step 5.

Author: Done.
